### PR TITLE
 [ddt] Change the console name to 'Dart Console' when in a Dart context

### PR DESCRIPTION
**Pull request by alan-knight**
Apologies for the extraneous formatting changes due to running VSCode's JS formatter.

### CODE REVIEW

**Comment by alan-knight** [comment]
Since it was Jacob's suggestion...

File: front_end/sdk/DartSupport.js [hunk 1]
```diff
@@ -5,6 +5,42 @@
...
+      SDK.DebuggerModel, SDK.DebuggerModel.Events.DebuggerPaused, this._debuggerPaused);
+  }
+
+  static async _isInDartContext() {
+    // TODO(alanknight): Reduce code duplication with $dartExpressionFor.
+    const executionContext = UI.context.flavor(SDK.ExecutionContext);
+    // We can't use selectedCallFrame() because it hasn't been set yet.
+    const selectedFrame = executionContext.debuggerModel.callFrames[0];
+    // We're not stopped at a breakpoint, treat it as JS.
+    if (!selectedFrame) return false;
```
Comment (jacob314) [inline front_end/sdk/DartSupport.js:22]: nit: why shouldn't the default be to treat it as dart? This tool is for Dart users.

File: front_end/sdk/DartSupport.js [hunk 2]
```diff
@@ -5,6 +5,42 @@
...
+      '',
+      false);
+    const enclosingLibraryName = await evaluation.currentLibrary();
+    // If there's an enclosing library name, then we're in a Dart context.
+    return enclosingLibraryName != null;
+  }
+
+  static async _debuggerPaused(event) {
+    const isDart = await Dart._NotificationHandler._isInDartContext();
+    const name = isDart ? 'Dart Console' : 'JavaScript Console';
```
Comment (jacob314) [inline front_end/sdk/DartSupport.js]: nit: 
const name = (isDart ? 'Dart' : 'JavaScript') + ' Console'

**Review by jacob314 — APPROVED** [review]

File: front_end/sdk/DartSupport.js (hunk 1 above)
Comment (alan-knight) [reply front_end/sdk/DartSupport.js:22]: It should, and there's a bug for that, but we need to do it differently if we don't have a frame to get the information from, and that isn't done yet.

File: front_end/sdk/DartSupport.js (hunk 2 above)
Comment (alan-knight) [reply front_end/sdk/DartSupport.js]: Done.

File: front_end/sdk/DartSupport.js (hunk 1 above)
Comment (jacob314) [reply front_end/sdk/DartSupport.js:22]: makes sense. maybe add a TODO to make the default dart in the future?

**Review by jacob314 — COMMENTED** [review]
lgtm

**Pull request merged**